demo/flutter/integration_test/app_test.dart

```diff
@@ -0,0 +1,128 @@
+
+  final String accessKey = "{TESTING_ACCESS_KEY_HERE}";
+  final String platform = Platform.isAndroid ? "android" : Platform.isIOS ? "ios" : "error";
+
+  Future<List<int>> loadAudioFile(String audioPath) async {
+      const INT16_MAX = 32767;
+      const INT16_MIN = -32768;
+
+      var audioFileData = await rootBundle.load(audioPath);
+      Wav audioFile = await Wav.read(audioFileData.buffer.asUint8List());
+      List<int> pcm = audioFile.channels[0].map((f) {
+          var i = (f * INT16_MAX).truncate();
+          if (f > INT16_MAX) i = INT16_MAX;
+          if (f < INT16_MIN) i = INT16_MIN;
+          return i;
+        }).toList();
+    return pcm;
+  }
+
+  group('Porcupine Keyword Tests', () {
+    testWidgets('Test singleKeyword all languages',
+      (tester) async {
+        String testDataJson = await rootBundle.loadString('assets/test_resources/test_data.json');
+        var testData = json.decode(testDataJson);
+
+        for (int t = 0; t < testData['tests']['singleKeyword'].length; t++) {
+          String language = testData['tests']['singleKeyword'][t]['language'];
+          String keyword = testData['tests']['singleKeyword'][t]['wakeword'];
+
+          String keywordPath = "assets/test_resources/keyword_files/${keyword}_${platform}.ppn";
+          String modelPath = "assets/test_resources/model_files/porcupine_params_$language.pv";
+          if (language == "en") {
+            modelPath = "assets/test_resources/model_files/porcupine_params.pv";
+          }
+
+          Porcupine porcupine;
+          try {
+             porcupine = await Porcupine.fromKeywordPaths(
+              accessKey,
+              [keywordPath],
+              modelPath: modelPath);
+          } on PorcupineException catch (ex) {
+            expect(ex, equals(null), reason: "Failed to initialize Porcupine for ${language}: ${ex}");
+            return;
+          }
+
+          String audioPath = "assets/test_resources/audio_samples/${keyword.replaceAll(' ', '_')}.wav";
+          List<int> pcm = await loadAudioFile(audioPath);
+
+          List<int> detections = [];
+          final int frameLength = porcupine.frameLength;
+          for (int i = 0; i < (pcm.length - frameLength); i += frameLength) {
+            int detection = await porcupine.process(pcm.sublist(i, i + frameLength));
+            if (detection != -1) {
+              detections.add(detection);
+            }
+          }
+
+          porcupine.delete();
+          expect(detections.length, equals(1), reason: "Number of detections for ${language} ${keyword} was incorrect");
+          expect(detections[0], equals(0), reason: "Porcupine returned wrong keyword index for ${language} ${keyword}");
+        }
+      });
+
+    testWidgets('Test multipleKeyword all languages',
+      (tester) async {
+        String testDataJson = await rootBundle.loadString('assets/test_resources/test_data.json');
+        var testData = json.decode(testDataJson);
```

Comment:
since this is used across multiple tests, we should load it once at the top/in a setup op. If that's not possible, a utility function will do.